lib/_stream_writable.js

```diff
@@ -384,20 +399,9 @@ function writeOrBuffer(stream, state, isBuf, chunk, encoding, cb) {
     state.needDrain = true;
 
   if (state.writing || state.corked) {
-    var last = state.lastBufferedRequest;
-    state.lastBufferedRequest = {
-      chunk,
-      encoding,
-      isBuf,
-      callback: cb,
-      next: null
-    };
-    if (last) {
-      last.next = state.lastBufferedRequest;
-    } else {
-      state.bufferedRequest = state.lastBufferedRequest;
-    }
-    state.bufferedRequestCount += 1;
+    const request = makeRequest(chunk, encoding, cb, isBuf);
+    state.bufferedRequests.push(request);
+    state.bufferedRequests.allBuffers &= isBuf;
```

Comment:
Is this a typo? Using a bitwise operator on a boolean property doesn't seem right.